parlai/crowdsourcing/tasks/model_chat/analysis/compile_results.py

```diff
@@ -0,0 +1,522 @@
+            [],
+            columns=[
+                'worker_id',
+                'conversations',
+                'problems_found',
+                'avg_problems_per_convo',
+            ],
+        )
+
+        for worker_id, data in worker_stats.items():
+            print(worker_id)
+            avg_problems_per_convo = data['problems_found'] / data['conversations']
+            stat = {
+                'worker_id': worker_id,
+                'conversations': data['conversations'],
+                'problems_found': data['problems_found'],
+                'avg_problems_per_convo': avg_problems_per_convo,
+            }
+            worker_df = worker_df.append(stat, ignore_index=True)
+        worker_df = worker_df.sort_values('avg_problems_per_convo', ascending=0)
+        worker_df.to_csv(worker_results_file, index=False)
+        print(worker_df)
+        print(f'Wrote worker statistical results to: {worker_results_file}')
+
+        html_text = (
+            '<html><body><table><tr><td>model</td>'
+            + ''.join(f'<td>{bucket}</td>' for bucket in self.regular_buckets)
+            + '<td>none_all_good</td><td>human_word_count</td><td>human_question_count</td><td>convo_clean</td><td>final_rating</td></tr>'
+        )
+        for model_nickname, model_problems_dict in problem_counts.items():
+            html_text += f'<tr><td>{model_nickname}</td>'
+            keys = self.regular_buckets + [
+                'none_all_good',
+                'human_word_count',
+                'human_question_count',
+            ]
+            for k in keys:
+                if k == 'human_word_count' or k == 'human_question_count':
+                    html_text += f'<td>{model_problems_dict[k]} ({model_problems_dict[k]/model_problems_dict["human_utterance_count"]:.3} average)</td>'
+                else:
+                    html_text += f'<td>{model_problems_dict[k]} ({model_problems_dict[k]/model_problems_dict["total"]:.1%})</td>'
+            html_text += f'<td>{model_problems_dict["convo_clean"]/model_problems_dict["count_convos"]:.1%} ({model_problems_dict["count_convos"]} convos)</td>'
+
+            html_text += f'<td>{np.average(model_problems_dict["ratings"]):.2f} ({model_problems_dict["count_ratings"]} ratings)</td>'
+
+            html_text += '</tr>'
+
+        html_text += '</table>'
+        print(html_text)
+
+        # Save full results
+        all_conversations_df = pd.DataFrame()
+        for df in conversation_dfs:
+            all_conversations_df = all_conversations_df.append(df)
+        all_conversations_df.to_csv(results_file, index=False)
+        print(f'\nWrote all conversation results to: {results_file}')
+        print(f'\nWorker conversation counts: {worker_conversation_counts}')
+
+        return all_conversations_df
+
+
+if __name__ == '__main__':
+    parser_ = ModelChatResultsCompiler.setup_args()
+    args_ = parser_.parse_args()
+    _ = ModelChatResultsCompiler(vars(args_)).compile_results()
```

Comment:
nit: can't you just call `ModelChatResultsCompiler(vars(args_)).compile_results()` without setting it to a return val?

Reply:
Fair enough, changed